### ISSUE TITLE
Web - Chat - Changed avatar is not updated at another user device until page refresh

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15082]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to https://staging.new.expensify.com/
and login on 2 devices with different accounts (let´s say account A and account B)
2. Create a chat between A and B
3. Account A - change avatar with a new photo
4. Account B - Click chat details with account A

## Expected Result:
After step 3 and 4 I would expect that account B see new avatar of account A

## Actual Result:
New avatar of account A is not displayed on account B device

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [X] iOS / native
- [X] iOS / Safari
- [x] Windows / Chrome
- [X] MacOS / Desktop

**Version Number:** 1.2.70.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/218271756-ebf55694-ea17-4d04-bb8b-62a4f6220762.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01248d0aa11dbe2e69</li>
        <li>Upwork Job ID: 1625926552308576256</li>
        <li>Last Price Increase: 2023-02-15</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by Christinadobrzyn** [comment]
I can replicate this following the steps in the OP. I think it might be related to https://github.com/Expensify/App/issues/10704

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcochavezf (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by marcochavezf** [comment]
Oh interesting, even though it seems related to the caching image project (https://github.com/Expensify/App/issues/10894) I think this is a backend issue where we're not sending and Pusher event to the other clients when an avatar is updated. I'm going to label it as internal atm and investigate a bit more.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01248d0aa11dbe2e69

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @rushatgabhane (`Internal`)

**Comment by hayata-suenaga** [comment]
@marcochavezf If you haven't started working on this, I'd like to take this issue to familiarize myself with the push notification part of web api. Please let me know what you think.

**Comment by marcochavezf** [comment]
Hi @hayata-suenaga! Sure, feel free to take it :)

**Comment by hayata-suenaga** [comment]
I'm working on PR for this issue now.

**Comment by hayata-suenaga** [comment]
This seems to be an issue in front end side since I confirmed the pusher event is sent by the backend. The avatar component might not be subscribing to the onyx data change properly.

**Comment by hayata-suenaga** [comment]
@marcochavezf would it be possible for us to work together on this issue? I feel like I'm stuck on investigating the cause of this issue.

When [`UpdateUserAvatar` API call](https://github.com/Expensify/Web-Expensify/blob/188e140b08cadaaffd84b0fd7b6feebdd0a241a7/api.php#L2352) is made, the corresponding method in UserAPI.php invokes [`pushUpdatesToClients`](https://github.com/Expensify/Web-Expensify/blob/188e140b08cadaaffd84b0fd7b6feebdd0a241a7/lib/UserAPI.php#L2796) of Onyx with correct value.

On the client side, however, the pusher event somehow doesn't trigger the Onyx data change callback [here](https://github.com/Expensify/App/blob/401cbca3a04782615b3c4bc8ee210e353d11396f/src/libs/SidebarUtils.js#L39).

When other pieces of public personal information is changed (ex. pronouns), the callback passed to the Onyx `connect` method is invoked correctly.

Could you give me a help in determining what other parts of the code I should look at to better search for the cause?

**Comment by Christinadobrzyn** [comment]
- Hired @rushatgabhane as the C+ for this internal job
- Internal Upwork job posting here - https://www.upwork.com/ab/applicants/1625926552308576256/job-details

**Comment by hayata-suenaga** [comment]
I'm still working on a PR for this issue

**Comment by MelvinBot** [comment]
@rushatgabhane, @Christinadobrzyn, @hayata-suenaga Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by hayata-suenaga** [comment]
waiting for another PR that addresses this issue to be deployed

**Comment by MelvinBot** [comment]
@rushatgabhane, @Christinadobrzyn, @hayata-suenaga Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by hayata-suenaga** [comment]
[The other PR that handles pusher event on avatar deletion has been merged to stating](https://github.com/Expensify/Web-Expensify/pull/36644#issuecomment-1462910967)

**Comment by Christinadobrzyn** [comment]
Looks like this just went into production so we'll give it 7 biz days before issuing payment.

**Comment by MelvinBot** [comment]
@rushatgabhane, @Christinadobrzyn, @hayata-suenaga Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Christinadobrzyn** [comment]
I think this has been 7 days in production without any regressions. @hayata-suenaga can you confirm we can pay out this job?

**Comment by hayata-suenaga** [comment]
@Christinadobrzyn PRs related to this issue were in backend. I think we only need to pay for the bug reporter.
